### PR TITLE
feat(api): v1: Add version specification to labware.load

### LINKED ISSUE #4216: api: apiv1: allow version specification of v2 labware
## overview

V2 labware definitions have associated versions. When we make substantive (geometry-affecting) changes to definitions, instead of changing the definition in place we add a new version of the definition. This version needs to be specified in the protocol when the labware was loaded. It is currently not possible to specify the labware version in apiv1. This hasn't been an issue up til now, but we are now adding new labware versions (in #4179 ) and the problem has become apparent.

## requested changes

Add a new keyword, `version=1`, to `labware.load`. This keyword is only valid when requesting a v2 labware by load name. When requesting a v2 labware, this version is used to determine the version of the labware def to load.

Also add significant documentation around this feature.

### PR DESCRIPTION
**Pull request by sfoster1**
V2 labware definitions have versions, which are bumped whenever the geometry
definitions change. It's important to be able to specify the version to load.
APIv2 allows this; the patch to add labware v2 loads from apiv1 does not. This
closes that gap.

Closes #4216

The code here is ready to review, but this PR shouldn't be merged until docs are in.

### CODE REVIEW

File: api/docs/source/v1/labware.rst [hunk 1]
```diff
@@ -99,6 +111,24 @@ deck, like modules. For this, you should use the ``share`` parameter.
...
                          share=True)
 
+To specify the version of the labware definition to use, you can use the ``version``
+parameter:
+
+.. code-block:: python
+
+   from opentrons import labware
+   block1 = labware.load(
+                'opentrons_96_aluminumblock_biorad_wellplate_200ul', 1,
```
Comment (mcous) [inline api/docs/source/v1/labware.rst]: Would this be easier to read if we use a named parameter for slot, too? Also, (actual change request) to be consistent with our docs, `slot` should be a string

```py
block1 = labware.load(
                'opentrons_96_aluminumblock_biorad_wellplate_200ul',
                slot='1',
                version=2)  # version 2 of the aluminum block definition
```

File: api/docs/source/v1/labware.rst [hunk 2]
```diff
@@ -33,10 +33,22 @@ If you are interested in using your own labware that is not included in the
...
+================
+
+Some labware on the Opentrons Labware Library have multiple versions of their
+definitions available. Opentrons publishes new versions of a labware definition
+when we find an issue with a labware definition. In general, you should use the
+newest version of a labware definition; however, the older definitions remain
+available for use with previously-written protocols that may have been customized
+to work with the older definition.
+
+
```
Comment (mcous) [inline api/docs/source/v1/labware.rst]: This section needs to spell out what version will be loaded if the version parameter is omitted

File: api/src/opentrons/legacy_api/api.py [hunk 3]
```diff
@@ -28,9 +28,11 @@ def create(self, *args, **kwargs):
...
         return cnt.list(*args, **kwargs)
 
-    def load(self, container_name, slot, label=None, share=False):
+    def load(
+            self, container_name, slot, label=None, share=False, version=None):
         try:
-            return cnt.load(self.robot, container_name, slot, label, share)
+            return cnt.load(
+                self.robot, container_name, slot, label, share, version)
         except FileNotFoundError:
```
Comment (mcous) [inline api/src/opentrons/legacy_api/api.py]: Do we raise any sort of `VersionNotFoundError` (or whatever) if the specified version is not found? If not, should we?

Comment (sfoster1) [reply api/src/opentrons/legacy_api/api.py]: There's nothing special, no - it's just another case of not finding the labware.

We probably should, and it should probably note which versions are available.

Comment (sfoster1) [reply api/src/opentrons/legacy_api/api.py]: The exception now mentions the version you passed. I think general prompting for available versions might want to wait.

**Pull request merged**